### ISSUE TITLE
Use entity IDs where possible instead of using the table URI

### ISSUE THREAD

**Issue by verdan** [lyft/amundsenmetadatalibrary#22]
For most of the part, amundsen metadata relies on the `table_uri`, which is being passed from frontend, that is making it complex and extra processing in case of systems which supports IDs. 
I am proposing to use the IDs of entities like Table, Column, and User etc to be used for the different kinds of operations. This information can be passed from metadata (when available, None otherwise), and simply be returned from frontend for each operation. 
Example Use Case: 
- Get Column Description:
At the moment, the frontend is sending the table_uri and the column name. I need to parse the table_uri, find table details, find the column based on the column name, and then get the column detail entity, and return the description. If there would be ID available for the column, then we can simply use that ID to get the record, and return the description.

Same is the case for Put, and Table detail itself. If we'd have the ID, we won't need to parse the table_uri. 

Just out of curiosity, how does Neo4j handles the IDs, does it support IDs ?

**Comment by bolkedebruin** [comment]
Cc @feng-tao

**Comment by jinhyukchang** [comment]
@verdan Neo4j supports ID but it's for internal purpose, not recommend for application to use.
Neo4j key (table URI for table node key) is defined in Databuilder and mostly shows its hierarchy. Thus,  column key can be easily derived from table key (table uri).

@verdan Could Atlas column can be derived from table key? Also, would you let me know where would be the proper place for column IDs to be available? (Don't think table detail API should ask for this and wonder where it can retrieve)

**Comment by verdan** [comment]
@jinhyukchang no, there is no way I can derive column ID from Table ID/ Table Key. The path to get column information from table_uri is already stated in the description of this issue. 

P.S: Atlas uses GUIDs as the entity IDs.

**Comment by bolkedebruin** [comment]
@verdan what about generating the qualifiedName from the URI? Ie qualifiedName for a column would be database.table.column@cluster which is indexed in atlas. Would that work?

**Comment by verdan** [comment]
@bolkedebruin that is a possibility, but in that case we need to make it a standard for all the users who want to connect atlas with amundsen to name their columns in this way. 
And, what about the existing data in atlas?

**Comment by jinhyukchang** [comment]
@verdan I can also see the benefit of using column entity ID as it can remove the knowledge of how it's being constructed from `metadata` service.

Table detail API to include column entity ID -> Frontend uses it when it calls column related API.

cc: @feng-tao

**Comment by Whazor** [comment]
I will adopt the pull request tomorrow to use `qualifiedName` without search and map the same URL structure to the same structure that Neo4J uses.

**Comment by L-KID** [comment]
@verdan Since frontend is using a single table 'key' in Metadata APIs, is it possible to give table GUID and column GUID to the 'key' for Atlas, and give table URI to the 'key' for Neo4j? So Atlas could use GUIDs to identify the table record in metadata service, while Neo4j still uses table URI.

**Comment by verdan** [comment]
@L-KID yes, that's the plan, but please keep in mind that frontend is using/PARSING the table key to navigate and make queries to metadata. So, if you are to pass the GUID as table key to frontend, you'll probably need to update the frontend to parse the table key accordingly. (Or maybe move the whole table key parsing to backend, to make it specific to the metadata proxy)

**Comment by feng-tao** [comment]
I think moving the parsing the logic the metadata proxy will be better to avoid FE handles different parsing logic.

**Comment by L-KID** [comment]
@verdan #56 with moving the parsing logic to metadata proxy.
